core/io/file_access_compressed.h

```diff
@@ -39,7 +39,7 @@ class FileAccessCompressed : public FileAccess {
 	bool writing = false;
 	uint64_t write_pos = 0;
 	uint8_t *write_ptr = nullptr;
-	uint32_t write_buffer_size = 0;
+	uint64_t write_buffer_size = 0;
```

Comment:
Sneaking some changes in here already? :P

Reply:
Making FileAccessCompressed call the 64-bit version is the motivation behind the rest of the changes, and is actually enough to make the function not crash for the test in Calinou's comment (to my surprise).